### DISCUSSION TITLE
Using prisma with serverless framework and webpack and typescript.

### DISCUSSION THREAD

**Discussion by jangrunicke** [prisma/prisma#15340] [Q&A] [answered]
I tried to use prisma in my serverless project, but I run into some problems. I use the serverless-webpack plugin to reduce the package size of the lambdas and use typescript at the same time. I already generated the prisma client, but I run into the following error message when invoking a function that uses the client to query the postgres data table: 

```
{
    "errorType": "Error",
    "errorMessage": "@prisma/client did not initialize yet. Please run \"prisma generate\" and try to import it again.\nIn case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues",
    "trace": [
        "Error: @prisma/client did not initialize yet. Please run \"prisma generate\" and try to import it again.",
        "In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues",
        "    at new PrismaClient (/var/task/node_modules/.prisma/client/index.js:3:11)",
        "    at Runtime.r [as handler] (/var/task/src/handler.js:1:583)",
        "    at Runtime.handleOnceNonStreaming (/var/runtime/Runtime.js:73:25)"
    ]
}
```

This is the function that gets called: 
```
export const getChildren = async (): Promise<APIGatewayProxyResult> => {
  const prisma = createPrismaClient();

  const children = await prisma.child.findMany();

  prisma.$disconnect;

  return {
    statusCode: 200,
    body: JSON.stringify(children, null, 2),
  } as ProxyResult;
};
```
This is the content of the file where I initialize my client:

`
import { PrismaClient } from "@prisma/client";

export const createPrismaClient = () => {
  return new PrismaClient();
};
`.

I think the issue is that the webpack bundler only sees the @primsa module as a dependency and does not pack the .prisma module as well. I tried to add this manually but failed to do so.

**Comment by nurul3101** [comment, marked as answer]
Hey @jangrunicke 👋 
What version of Prisma are you using?

Did you had a look at our Deployment using the [serverless-webpack](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-aws-lambda#deployment-using-serverless-webpack) guide?
You can use the [serverless-webpack-prisma](https://github.com/danieluhm2004/serverless-webpack-prisma) plugin without needing to do any additional configuration.

**Reply by pantharshit00** [reply]
Hey @jangrunicke, Were you able to look into the about resources?

**Reply by jangrunicke** [reply]
Hey, sorry for the late reply. Yes I have fixed my issue through the ressources provided by @nurul3101 thank you very much.

**Comment by nmfrankel** [comment]
I recently ran into this issue, modify/add the build command to `"vercel-build": "prisma generate && run build"`.

My project is hosted on Vercel, hence the `vercel-build` prefix, for greater detail you can check out my [package.json](https://github.com/nmfrankel/ChatService/blob/main/package.json)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
